lib/mix/tasks/archive/build.all.ex

```diff
@@ -61,7 +61,7 @@ defmodule Mix.Tasks.Archive.Build.All do
     destination = Mix.Archive.Build.Helpers.destination(opts)
     Mix.Tasks.Archive.Build.Deps.build_archives(opts)
 
-    archive_name = Mix.Local.name_for(:archive, Mix.Project.config)
+    archive_name = Mix.Local.name_for(:archives, Mix.Project.config)
```

Comment:
This was changed in commit https://github.com/elixir-lang/elixir/commit/a94911881d4c0424fe3eaf154aa0d87cc5163b51. If we want to keep backwards compatibility, we need something a bit more elegant.

Reply:
We don't build with anything earlier than Elixir 1.9, I believe.

Reply:
Even with 1.8.2 we don't use `mix_task_archive_deps` anymore.

(cc @dumbbell @michaelklishin)

Reply:
Dropping Elixir 1.8 support is fine with me.